database/database.go

```diff
@@ -105,6 +107,57 @@ func (db *Database) RecordRequestToJoinTeam(
 	return db.saveToFile(*message)
 }
 
+// RecordLast takes a verb and item and records the action in the database, e.g verb "fetched",
+// item: key.
+func (db *Database) RecordLast(verb string, item interface{}, now time.Time) error {
+	message, err := db.loadFromFile()
+	if err != nil {
+		return err
+	}
+
+	switch i := item.(type) {
+	case *pgpkey.PgpKey:
+		message.LastUpdated[verb+":"+keyItem+":"+i.Fingerprint().Uri()] = now
+	case pgpkey.PgpKey:
+		message.LastUpdated[verb+":"+keyItem+":"+i.Fingerprint().Uri()] = now
+	case *fpr.Fingerprint:
+		message.LastUpdated[verb+":"+keyItem+":"+i.Uri()] = now
+	case fpr.Fingerprint:
+		message.LastUpdated[verb+":"+keyItem+":"+i.Uri()] = now
+	case *team.Team:
+		message.LastUpdated[verb+":"+teamItem+":"+i.UUID.String()] = now
+	case team.Team:
+		message.LastUpdated[verb+":"+teamItem+":"+i.UUID.String()] = now
+	}
+
+	return db.saveToFile(*message)
+}
+
+// GetLast akes a verb and item and returns the last time it was recorded as being 'done'.
+func (db *Database) GetLast(verb string, item interface{}) (lastUpdated time.Time, err error) {
+	message, err := db.loadFromFile()
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	switch i := item.(type) {
+	case *pgpkey.PgpKey:
+		return message.LastUpdated[verb+":"+keyItem+":"+i.Fingerprint().Uri()], nil
+	case pgpkey.PgpKey:
+		return message.LastUpdated[verb+":"+keyItem+":"+i.Fingerprint().Uri()], nil
+	case *fpr.Fingerprint:
+		return message.LastUpdated[verb+":"+keyItem+":"+i.Uri()], nil
+	case fpr.Fingerprint:
+		return message.LastUpdated[verb+":"+keyItem+":"+i.Uri()], nil
+	case *team.Team:
+		return message.LastUpdated[verb+":"+teamItem+":"+i.UUID.String()], nil
+	case team.Team:
+		return message.LastUpdated[verb+":"+teamItem+":"+i.UUID.String()], nil
+	}
+
+	return time.Time{}, fmt.Errorf("no record of when %v was last '%s'", item, verb)
```

Comment:
it would be helpful for my use case to tell the difference between an error and "everything OK, but key not found".

suggest a custom error type `ErrNotFound`